File: Data Structures & Algorithms/confusing-number/submission-0.cpp
class Solution {
public:
    bool confusingNumber(int n) {
        map<int, int> invertMap = {{0, 0}, {1, 1}, {6, 9}, {8, 8}, {9, 6}};
        int rotatedNumber = 0, nCopy = n;

        while (nCopy > 0) {
            int res = nCopy % 10;
            if (invertMap.find(res) == invertMap.end()) {
                return false;
            }

            rotatedNumber = rotatedNumber * 10 + invertMap[res];
            nCopy /= 10;
        }
        
        return rotatedNumber != n; 
    }
};

File: Data Structures & Algorithms/pow-x-n/submission-0.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        res = 1
        neg = False
        if n < 0:
            n *= -1
            neg = True
        while n > 0:
            if n&1:
                res *= x
            x *= x
            n = n>>1
        return 1/res if neg else res

        